File: URI/1279 Leap Year or Not Leap Year and …/main.cpp
#include<bits/stdc++.h>

using namespace std;

int mod(string s, int mod){
    int rslt = 0;

    for(int i=0; i<s.length(); i++){
        rslt = ((rslt * 10) + (s[i] - '0'))%mod;
    }

    return rslt%mod;
}

int main(){

    string year;

    bool first = false;
    while(cin>>year){

        if(first) cout<<endl;

        bool festival = false;
        bool leap = false;
        first = true;

        if (((mod(year, 4) == 0) && (mod(year, 100) != 0)) || (mod(year, 400) == 0)){
            cout<<"This is leap year."<<endl;
            festival = true;
            leap = true;
        }

        if (mod(year, 15) == 0){
            cout<<"This is huluculu festival year."<<endl;
            festival = true;
        }

        if (mod(year, 55) == 0 &&  leap == true){
            cout<<"This is bulukulu festival year."<<endl;
        }

        if (festival == false){
            cout<<"This is an ordinary year."<<endl;
        }


    }

    return 0;
}

File: URI/1279 Leap Year or Not Leap Year and …/main.py
from sys import stdin

first = False

for line in stdin:
    year = int(line)
    
    if (first == True):
        print("")

    festival = False
    leap = False
    first = True

    if (((year % 4 == 0) and (year % 100 != 0)) or (year % 400 == 0)):
        print("This is leap year.")
        festival = True
        leap = True

    if (year%15 == 0) :
        print("This is huluculu festival year.")
        festival = True

    if (year%55 == 0 and  leap == True):
        print("This is bulukulu festival year.")

    if festival == False:
        print("This is an ordinary year.")
    